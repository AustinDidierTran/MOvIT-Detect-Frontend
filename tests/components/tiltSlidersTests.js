/**
 * @author Austin Didier Tran
 */

import React from 'react';
import TestUtils from 'react-dom/test-utils';
import PropTypes from 'prop-types';
import TiltSliders from '../../src/components/tiltSliders';

describe('TiltSliders Tests', () => {
  it('should have proptypes', function () {
    // Actual value
    const actualValue = TiltSliders.WrappedComponent.propTypes;

    // Expected value
    const expectedValue = {
      language: PropTypes.string.isRequired,
      tiltFrequecy: PropTypes.number,
      tiltLength: PropTypes.number,
      tiltAngle: PropTypes.number,
      maxAngle: PropTypes.number,
      onFrequencyChange: PropTypes.func.isRequired,
      onLengthChange: PropTypes.func.isRequired,
      onAngleChange: PropTypes.func.isRequired,
    };

    // Test
    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });
});
