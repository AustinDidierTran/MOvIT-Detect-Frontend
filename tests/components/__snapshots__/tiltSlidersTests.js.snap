// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['TiltSliders Tests should match the snapshot 1'] = `
<TiltSliders
  dispatch={[Function]}
  language="en"
  maxAngle={45}
  onAngleChange={[Function]}
  onFrequencyChange={[Function]}
  onLengthChange={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tiltAngle={15}
  tiltFrequecy={60}
  tiltLength={30}
/>
`;
